build.gradle

```diff
@@ -64,17 +64,17 @@ subprojects {
 
         errorprone 'com.google.errorprone:error_prone_core:2.3.1'
 
-        testCompile 'nl.jqno.equalsverifier:equalsverifier:2.4.5'
+        testCompile 'nl.jqno.equalsverifier:equalsverifier:2.5.2'
         testCompile "org.hamcrest:java-hamcrest:$hamcrestVersion"
+        testCompile "org.junit.jupiter:junit-jupiter-api:$junitJupiterVersion"
+        testCompile 'org.junit-pioneer:junit-pioneer:0.1.2'
         testCompile "org.mockito:mockito-core:$mockitoVersion"
         testCompile "org.mockito:mockito-junit-jupiter:$mockitoVersion"
 
         testCompileOnly "org.projectlombok:lombok:$lombokVersion"
 
-        testImplementation "org.junit.jupiter:junit-jupiter-api:$junitJupiterVersion"
```

Comment:
The `*Implementation` and `*Api` configurations are the new Gradle way to ensure implementation dependencies are not leaked outside of a module.  However, since this is the only place where we use such a configuration, I decided to change it to its old-style equivalent (`testCompile`) for consistency.

As we're becoming more modular, in the future, we should go through and convert all `*Compile` dependencies to either `*Api` or `*Implementation`, as appropriate.

Reply:
I'm a little bit 'meh' about this. 
If you ask me, leaving this as it is is just fine as it's some sort of progress in this, but not that important.
What's more important though is the api dependencies. Was confused about this as well when I read about it quite some time ago, so I double checked with the current [gradle docs of the java plugin](	https://docs.gradle.org/current/userguide/java_plugin.html#tab:configurations).
It seems that the `compile` dependency declaration is now deprecated, and `implementation` should be used instead (as well as `runtime` which should be replaced by `runtimeOnly`).
But at no place `api` is mentioned except for one place where they reffer to the `java-library` plugin which has slightly different mechanics including the `api` declaration you mentioned, so using this one would require us to switch the plugins as we're not really deploying a library and are really just shipping the application as a whole I really don't think we need `api` at all, especially because `implementation` should be preferred anyways.

Reply:
> If you ask me, leaving this as it is is just fine as it's some sort of progress in this, but not that important.

744b71474

> But at no place api is mentioned except for one place where they reffer to the java-library plugin which has slightly different mechanics including the api declaration you mentioned, so using this one would require us to switch the plugins as we're not really deploying a library and are really just shipping the application as a whole I really don't think we need api at all, especially because implementation should be preferred anyways.

Technically, `test-common` and `game-core` (and possibly `http-data`) are _internal_ libraries and have an API they expose to downstream modules.  If, for example, `game-core` were to use a third-party library type in a public method signature that is called by a downstream module, that dependency should be marked `api`.  I'm pretty sure if you marked it `implementation`, you'd get a compiler error because the third-party type wouldn't be visible outside the source module (unless, of course, the downstream module _also_ declared that dependency).  The benefit of `api` is it extends the dependency transitively to downstream modules.

With that said, however, I agree that all of our dependencies should default to `implementation`, and we should only switch it to `api` (in an internal library module) if it's actually used in a public API.

Reply:
In this case the `java` plugin could be replaced just in those sub projects.

Reply:
Yup.